torch/csrc/distributed/c10d/comm.hpp

```diff
@@ -21,12 +21,14 @@ class TORCH_API GradBucket {
       const at::Tensor& tensor,
       const std::vector<size_t>& offsets,
       const std::vector<size_t>& lengths,
-      const std::vector<c10::IntArrayRef>& sizes_vec)
+      const std::vector<c10::IntArrayRef>& sizes_vec,
+      const std::unordered_map<size_t, at::Tensor>& bucket_idx_to_variable)
       : index_(index),
         tensor_(tensor),
         offsets_(offsets),
         lengths_(lengths),
-        sizes_vec_(sizes_vec) {}
+        sizes_vec_(sizes_vec),
+        bucket_idx_to_variable_(bucket_idx_to_variable) {}
```

Comment:
could we just call getVariables() to return as list of variables? as the idx is the same as idx in getPerParameterTensor() anyway?